rplugin/python3/deoplete/deoplete.py

```diff
@@ -53,6 +55,13 @@ def __init__(self, vim):
         self.__profile_start = 0
 
     def completion_begin(self, context):
+        if not logger.init:
+            log_file = None
+            if self.__vim.vars['deoplete#enable_debug']:
+                self.debug_enabled = True
+                log_file = self.__vim.vars['deoplete#debug_log']
+            logger.setup(self.__vim, log_file)
```

Comment:
> Why logger.setup is g:deoplete#enable_debug even false?

Mainly to set `logger.init` to `True` since I think that the logger's state variable should be managed internally.  If `self.__vim.vars['deoplete#enable_debug']` is checked each time, that would cause a slowdown (I think).  Something needs to know that `enable_debug` was already checked.

I feel that once this is added, it will be the least updated file in the repo.  I wanted to keep changes to the core minimal.  That was also the reason why I originally placed it in `__init__.py`

> Also, I think not necessary root.addHandler(logging.NullHandler())

I have a habit of being explicit if the cost is low.  I have no problem with removing this if you think it has no benefit.